### PR TITLE
Fix failing unit test

### PR DESCRIPTION
**Pull request by MarkEWaite**
# Fix unit test and deprecations

* Correct assertion for current code
* Use non-deprecated jsoup SafeList method
* Use String.join instead of hudson.Util.join

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -62,8 +62,8 @@
     <dependencies>
       <dependency>
         <groupId>io.jenkins.tools.bom</groupId>
-        <artifactId>bom-2.263.x</artifactId>
-        <version>961.vf0c9f6f59827</version>
+        <artifactId>bom-2.289.x</artifactId>
```
Comment (timja) [inline pom.xml:65]: Shouldn’t we use the weekly bom here?

**Review by kwhetstone — APPROVED** [review]

**Comment by timja** [comment]
Is the test actually failing? changes here look good but this went green with no changes:
https://github.com/jenkins-infra/pipeline-steps-doc-generator/pull/111

**Comment by MarkEWaite** [comment]
> Is the test actually failing? changes here look good but this went green with no changes: #111

Test was failing in my local development environment, but I'll need to check again.  I'll place this PR in Draft until I have time to check it.

**Comment by MarkEWaite** [comment]
The test fails for me when I use the development instructions in the ["Create content" section](https://github.com/jenkins-infra/pipeline-steps-doc-generator#create-content) of the README.

The test fails for me when I run `mvn clean verify` in the root of this repository from the master branch.  I also had to add an additional repository to my `~/.m2/settings.xml` to find the `ua_parser` from the Twitter maven repository.  I know that I've done development work in this repository previously, but I seem to have lost the recipe and am now bumping into issues.

**Review by timja — APPROVED** [review]

**Comment by MarkEWaite** [comment]
I'm leaving this in Draft state until the www.jenkins.io build and development process is fixed.  The test passes on this branch but when I try to deploy a development copy, it fails in the www.jenkins.io build process.  I suspect that is the same failure we're seeing on the www.jenkins.io master branch

**Comment by MarkEWaite** [comment]
I downloaded the allAscii.zip before this change and the one after this change.  Comparison shows only two files with differences.

Before the change docker-plugin.adoc included a line for `dockerNode` that it listed as experimental.  After the change, that line was removed.  The docker plugin 1.2.6 was released just 2 hours ago, so I believe a change in that location might be explained.  No [changelog entry](https://github.com/jenkinsci/docker-plugin/blob/master/CHANGELOG.md) yet for 1.2.6,.

Before the change, docker-slaves.adoc did not exist.  Now it does.  The docker-slaves last release was 4 years ago.  I believe the inclusion of that in the documentation is correct.  The [plugin](https://plugins.jenkins.io/docker-slaves/) is available and does not have any security issues flagged on the plugins page.

**Pull request merged**